src/interactions/slashCommands/mixins/ApplicationCommandOptionChannelTypesMixin.ts

```diff
@@ -0,0 +1,55 @@
+import { ChannelType } from 'discord-api-types/v9';
+import { z, ZodLiteral } from 'zod';
+
+// Only allow valid channel types to be used. (This can't be dynamic because const enums are erased at runtime)
+const allowedChannelTypes = [
+	ChannelType.GuildText,
+	ChannelType.GuildVoice,
+	ChannelType.GuildCategory,
+	ChannelType.GuildNews,
+	ChannelType.GuildStore,
+	ChannelType.GuildNewsThread,
+	ChannelType.GuildPublicThread,
+	ChannelType.GuildPrivateThread,
+	ChannelType.GuildStageVoice,
+] as const;
+
+export type ApplicationCommandOptionAllowedChannelTypes = typeof allowedChannelTypes[number];
+
+const channelTypePredicate = z.union(
+	allowedChannelTypes.map((type) => z.literal(type)) as [
+		ZodLiteral<ChannelType>,
+		ZodLiteral<ChannelType>,
+		...ZodLiteral<ChannelType>[]
+	],
+);
+
+export class ApplicationCommandOptionChannelTypesMixin {
```

Comment:
So I was messing around with this, and it seems that the channel option doesn't need to be a mixin (since it's only used in one place), this code works on it's own:

```ts
export class ApplicationCommandOptionChannelTypesMixin extends ApplicationCommandOptionBase<ApplicationCommandOptionType.Channel> {
	public override readonly type = ApplicationCommandOptionType.Channel as const;

  ...

  public toJSON(): APIApplicationCommandChannelOption {
    this.runRequiredValidations();

    return { ...this };
	}
}
```

Reply:
While true, I'd keep it as a mixin for future proofing